protocols/ledger/ledger.go

```diff
@@ -23,55 +23,42 @@ func NewLedgerManager() LedgerManager {
 // It returns a signed state message that can be sent to other participants.
 func (l *LedgerManager) HandleRequest(ledger *channel.TwoPartyLedger, request protocols.LedgerRequest, secretKey *[]byte) (protocols.SideEffects, error) {
 
-	guarantee, _ := outcome.GuaranteeMetadata{
-		Left:  request.Left,
-		Right: request.Right,
-	}.Encode()
-
 	supported, err := ledger.Channel.LatestSupportedState()
 	if err != nil {
 		return protocols.SideEffects{}, fmt.Errorf("error finding a supported state: %w", err)
 	}
-
-	asset := types.Address{} // todo: loop over request.amount's assets
 	nextState := supported.Clone()
 
-	// Get the current amounts from the ledger channel
-	currentLeftAmount := nextState.Outcome.TotalAllocatedFor(request.Left)[asset]
-	currentRightAmount := nextState.Outcome.TotalAllocatedFor(request.Right)[asset]
-	// Calculate the new amounts by subtracting the requested amounts from the current amounts
-	leftAmount := big.NewInt(0).Sub(currentLeftAmount, request.LeftAmount[asset])
-	rightAmount := big.NewInt(0).Sub(currentRightAmount, request.RightAmount[asset])
+	for i, exit := range nextState.Outcome {
+		asset := exit.Asset
+		// If our request doesn't deal with this asset, skip it
+		if types.IsZero(request.LeftAmount[asset]) && types.IsZero(request.RightAmount[asset]) {
+			continue
+		}
```

Comment:
Is this part unit tested? If the request doesn't deal with an asset in the `nextState.Outcome`, I think this attempted access may result in a panic instead of a zero amount being returned?